# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 6c129bd72b61
# Full Hash: 6c129bd72b61cf6d096dcec6c7bb188e3ddb37c7
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-15 09:27:59
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement InitModuleLoader methods for WorkerScriptLoader and WorkerGlobalScope; r=jonco
#   
#   This implements a method to initialize the moduleLoader for workers. This will initialize only once, for all worker types (module and classic).
#   
#   Depends on D147324
# ==============================================================================

diff -r 4b0a4fcc6894 -r 6c129bd72b61 dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Wed Dec 14 14:55:34 2022 +0000
+++ b/dom/workers/WorkerScope.h	Wed Dec 14 14:55:35 2022 +0000
@@ -8,6 +8,7 @@
 #define mozilla_dom_workerscope_h__
 
 #include "js/TypeDecls.h"
+#include "js/loader/ModuleLoaderBase.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/Attributes.h"
 #include "mozilla/DOMEventTargetHelper.h"
@@ -158,8 +159,23 @@
 
   Console* GetConsoleIfExists() const { return mConsole; }
 
+  void InitModuleLoader(JS::loader::ModuleLoaderBase* aModuleLoader) {
+    if (!mModuleLoader) {
+      mModuleLoader = aModuleLoader;
+    }
+  }
+
+  // The nullptr here is not used, but is required to make the override method
+  // have the same signature as other GetModuleLoader methods on globals.
+  JS::loader::ModuleLoaderBase* GetModuleLoader(
+      JSContext* aCx = nullptr) override {
+    return mModuleLoader;
+  };
+
   uint64_t WindowID() const;
 
+  virtual void NoteTerminating();
+
   // Usually global scope dies earlier than the WorkerPrivate, but if we see
   // it leak at least we can tell it to not carry away a dead pointer.
   void NoteWorkerTerminated() { mWorkerPrivate = nullptr; }
@@ -184,6 +200,7 @@
 
  private:
   RefPtr<Console> mConsole;
+  RefPtr<JS::loader::ModuleLoaderBase> mModuleLoader;
   const UniquePtr<ClientSource> mClientSource;
   nsCOMPtr<nsISerialEventTarget> mSerialEventTarget;
   bool mShouldResistFingerprinting;
@@ -217,7 +234,7 @@
 
   using WorkerGlobalScopeBase::WorkerGlobalScopeBase;
 
-  void NoteTerminating();
+  void NoteTerminating() override;
 
   void NoteShuttingDown();
 